provider/openstack/provider.go

```diff
@@ -1001,9 +1001,16 @@ func (e *Environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	networkId, err := e.networking.ResolveNetwork(usingNetwork, false)
 	if err != nil {
 		if usingNetwork == "" {
-			err = errors.New(noNetConfigMsg(err))
+			// If there is no network configured, we only throw out when the
+			// error reports multiple Openstack networks.
+			// If there are no Openstack networks at all (such as Canonistack),
+			// having no network config is not an error condition.
+			if strings.HasPrefix(err.Error(), "multiple networks") {
```

Comment:
Is there any way to have a better analysis of the error than checking the string starts with "multiple networks"?
I like the idea of this, but where does that error originate from?
How stable is that string going to be? Is there any chance there is an error code we can recognize/have Goose translate an obvious error into an error code?

Reply:
In this case we control that error via `processResolveNetworkIds` (in _networking.go_). I considered a custom error type, but since it's in our code, I figured having coverage keeps us safe.